pkg/services/oauthtoken/oauth_token_test.go

```diff
@@ -270,3 +150,382 @@ func (f *FakeAuthInfoStore) UpdateAuthInfo(ctx context.Context, cmd *login.Updat
 func (f *FakeAuthInfoStore) DeleteAuthInfo(ctx context.Context, cmd *login.DeleteAuthInfoCommand) error {
 	return f.ExpectedError
 }
+
+func TestService_TryTokenRefresh(t *testing.T) {
+	type environment struct {
+		authInfoService *authinfotest.FakeService
+		cache           *localcache.CacheService
+		identity        identity.Requester
+		socialConnector *socialtest.MockSocialConnector
+		socialService   *socialtest.FakeSocialService
+		validations     func()
+
+		service *Service
+	}
+	type testCase struct {
+		desc        string
+		expectedErr error
+		setup       func(env *environment)
+	}
+
+	tests := []testCase{
+		{
+			desc: "should skip sync when identity is nil",
+		},
+		{
+			desc: "should skip sync when identity is not a user",
+			setup: func(env *environment) {
+				env.identity = &authn.Identity{ID: "service-account:1"}
+			},
+		},
+		{
+			desc: "should skip token refresh and return nil if namespace and id cannot be converted to user ID",
+			setup: func(env *environment) {
+				env.identity = &authn.Identity{ID: "user:invalidIdentifierFormat"}
+			},
+		},
+		{
+			desc: "should skip token refresh if RefreshToken is empty",
+			setup: func(env *environment) {
+				env.authInfoService.ExpectedUserAuth = &login.UserAuth{}
+
+				env.identity = &user.SignedInUser{
+					AuthenticatedBy: login.GenericOAuthModule,
+					UserID:          1,
+				}
+
+				env.socialService.ExpectedAuthInfoProvider = &social.OAuthInfo{
+					UseRefreshToken: true,
+				}
+			},
+		},
+		{ /** TestService_TryTokenRefresh_ValidToken **/
+			desc: "should skip token refresh since the token is still valid",
+			setup: func(env *environment) {
+				token := &oauth2.Token{
+					AccessToken:  "testaccess",
+					RefreshToken: "testrefresh",
+					Expiry:       time.Now(),
```

Comment:
You need to add 1 hour if you want the token to still be valid.
```suggestion
					Expiry:       time.Now().Add(time.Hour),
```